packages/core/src/interpreter.ts

```diff
@@ -296,12 +293,10 @@ export class Interpreter<
     registry.register(this.sessionId, this.ref);
     this.status = ActorStatus.Running;
 
-    let resolvedState = initialState
-      ? this.behavior.restoreState?.(initialState, this._actorContext)
-      : this._getInitialState() ?? undefined;
+    let resolvedState = this._getInitialState();
 
     if (this.behavior.start) {
-      resolvedState = this.behavior.start(resolvedState, this._actorContext);
+      this.behavior.start(resolvedState, this._actorContext);
```

Comment:
is it a change in the contract between the `interpret` and the `behavior`? The `ActorBehavior` type still mentions `TInternalState` as the return type from `start`:
https://github.com/statelyai/xstate/blob/3769caea411bf7a4390ae34a4457c6bc7a464afe/packages/core/src/types.ts#L1837-L1840

Reply:
This should be changed; `.start(...)` is too late to provide any state.

Reply:
It could still return a new state - based on the provided one. But I agree, it's best if the whole initial state is known upfront. We don't want to create cascading updates on mount etc (even if that's only a UI concern). 

So ye - this signature should be probably changed